### DISCUSSION TITLE
yeartoyear behavior

### DISCUSSION THREAD

**Discussion by chankeith** [davidquinn/psi-header#11] [Q&A]
I have a question on how to best configure a certain behavior.  Perhaps `yeartoyear` already does this, but I just want to be sure since I can't seem to fake an older creation date on Linux.

I want to have a copyright year range.  The "from" year is certainly the original creation year, and the "to" year is the current year.  Would template substitution update the "to" year if the current year has changed?  For example, would 2015-2021 change to 2015-2022?

What can I do if I wanted to retain the previous years and append the current year when the file has been updated, such as 2015, 2018, 2019, 2021?

**Comment by davidquinn** [comment]
Hi,

Yes, using `yeartoyear(fc, now)` would do what you are asking - `from` would always be the file creation year and `to` would always be the year that the file was last saved within Visual Studio Code.

So if you created the file in 2015 and edited it in 2021 it will show "2015 - 2021". Then if you edited it again in 2022 it would change to "2015 - 2022".

As for having a list of years, the only way to support that would be via the extension's [Changes Log](https://github.com/davidquinn/psi-header#change-log) feature to create a log entry whenever a file is changed.

HTH
David

**Reply by chankeith** [reply]
Thanks.  This helps.

Would cloning the repository and subsequently updating any file affect the file's creation year in the header since the file was "created" in a sense through the cloning process?

**Reply by davidquinn** [reply]
Yes, it is going to treat whatever the OS is reporting as the creation date which for a cloned repository is when the file was cloned. Assuming your project starts from a specified time, you could always use a static string as the from year.

HTH
D
